tool/update_github_emojis.dart

```diff
@@ -0,0 +1,338 @@
+  '3030',
+  '2714',
+  '2716',
+  '00a9',
+  '00ae',
+  '2122',
+  '2611',
+  '25aa',
+  '25ab',
+  '25fc',
+  '25fb',
+  '2660',
+  '2663',
+  '2665',
+  '2666',
+];
+
+const errorSpecialReplacement = '\u{FFFD}'; // Special replacement character '�'
+
+const useOfGithubUnicodeSequencesWarning = '''
+IMPORTANT NOTE: The use of the --useGithubUnicodes switch will force using
+Github unicode sequences.
+This option is essentially here only for completeness, not for
+release use.
+The slight visual differences of some emojis might also be another
+reason using --useGithubUnicodes should be considered a *Breaking Change*.
+
+Some test will fail because of the different unicode sequences
+and the emojis.unit file would need to be updated to contain the new
+expected Github versions of the unicode sequences of the emojis in order
+for the tests to pass.
+''';
+
+/// The github api url will return a json map of alls emojis in form of
+/// { 'shortcode':'emojifilename' ... }
+/// The filenames are simply a list of all of the hex string of the
+/// *essential* unicode codepoints representing the emoji.
+/// These sequences exclude the unicode join zero width (0x200D) and
+/// variation select (0xFE0F) modifiers. (We will need to add these in to
+/// build our actually unicode strings representing the emoji).
+/// Multiple unicode codepoints are separated by '-'.
+/// Examples filenames (single and double code point examples):
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f643.png?v8"
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f1fa-1f1fe.png?v8"
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f469-1f469-1f467-1f466.png?v8"
+/// NOTE: Some filenames will be github 'custom' emojis that have no unicode
+/// equivalent and these will not have hex codepoints, only the github custom name.
+/// We will ingore these (there are only a 19 and they are mostly pixel art from
+/// the old Doom game).
+/// Example github custom emoji filename:
+///  - "https://github.githubassets.com/images/icons/emoji/godmode.png?v8",
+String parseGithubFilenameIntoUnicodeString(String emojiFilename) {
+  const variationSelector = 0xFE0F;
+  const zeroWidthJoiner = 0x200D;
+
+  try {
+    final String? rawHexList =
+        parseGithubEmojiUnicodeFromFilename.firstMatch(emojiFilename)?.group(1);
+    if (rawHexList == null) {
+      // This is a github custom emoji and it is represent by PNG only and
+      // there is no equivalent unicode.  We have to ingore.
+      return '';
+    }
+    bool legacyUsedVariationCode = false;
+    if (legacyEmojisUsedVariationModifier.contains(rawHexList)) {
+      legacyUsedVariationCode = true;
+    }
+    final List<int> rawCodePointsHex = rawHexList
+        .split('-')
+        .map((hexstr) => int.parse(hexstr, radix: 16))
+        .toList();
+    final List<int> codePointsHex = [];
+
+    if (legacyUsedVariationCode) {
+      // just add single variation selector
```

Comment:
Nit: Capitalize and end with a period: https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences

Reply:
I don't think this has been addressed.